### PR TITLE
feat: 360 image annotation styling

### PR DESCRIPTION
**Pull request by haakonflatval-cognite**
#### Type of change

Feat  

#### Jira ticket :blue_book:

https://cognitedata.atlassian.net/browse/REV-876

## Description :pencil:

Adds simple styling capabilities to annotations. It sort-of mimics cad and point cloud models, where the `Image360Collection` takes the place of the model. 

A big difference is that styling filters is a predicate function, rather than a "filter" object. This is mostly due to the sheer amount of data in the `AnnotationModel` object describing the annotation which may be relevant, so creating a flexible, exhaustive API based on "filter objects" may be hard. There will also be a much smaller performance hit than what we had encountered using predicate functions for node / point cloud object styling, since we only need to consider the annotations associated with the current 360 revision when applying styles.

This design is up for discussion, however.

## How has this been tested? :mag:

- Start Reveal examples
- Open https://localhost:3000/?project=3d-test&env=greenfield-3d&modelId=4923307033982497&revisionId=3100284276970593
- Load 360 image set `helideck-site-2-jpeg`
- Enter the lower-most 360-entity (has the most annotations)
- Observe that some annotations are differently colored, according to lines 178-182 in `Image360UI`
- Observe that clicking an annotation changes its color so that it feels "selected"
- Observe that selection changes when clicking new annotation
- Observe that adding a style using the new "360 annotation styling" UI colors/hides the annotations matching the label text given (the console lists the label of clicked annotations).

## Checklist :ballot_box_with_check:

- [x] I am proud of this feature.
- [ ] I have performed a self-review of my own code.
- [x] I have added PR type (Feat, Bug, Chore, Test, Docs, Style, Refactor) to the PR title.
- [ ] I have manually tested this for different scenarios (different models, formats, devices, browsers).
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the public documentation.
- [ ] I have added unit and visuals tests to prove that my feature works to the best of my ability.
- [ ] I have refactored the code for readability to the best of my ability.
- [ ] I have checked that my changes do not introduce regressions in the public documentation.
- [ ] I have outlined any known defects / lacking capabilities in the contents of this PR.
- [ ] I have listed any remaining work that should follow this PR in the description or jira/miro/etc.
- [ ] I have added TSDoc to any public facing changes.
- [ ] I have added "developer documentation" in any package README.md that is related / applicable to this PR.
- [ ] I have noted down and am currently tracking any technical debt introduced in this PR.
- [ ] I have thoroughly thought about the architecture of this implementation.
- [ ] I have asked peers to test this feature.

### CODE REVIEW

File: viewer/packages/360-images/src/annotation-styling/AnnotationStyleTracker.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+ */
+
+import { AnnotationModel } from '@cognite/sdk/dist/src';
+import { ImageAnnotationObject } from '../annotation/ImageAnnotationObject';
+import { Image360AnnotationAppearance, Image360AnnotationAppearanceEdit, Image360AnnotationFilter } from './types';
+
+import { Color, MeshBasicMaterial } from 'three';
+
+export class AnnotationStyleTracker {
+  private readonly _annotationStyleAssignments: [Image360AnnotationFilter, Image360AnnotationAppearanceEdit][] = [];
```
Comment (christjt) [inline viewer/packages/360-images/src/annotation-styling/AnnotationStyleTracker.ts:12]: Maybe this should be a map? Just seems more ergonomic based on how the collection is used.

File: viewer/packages/360-images/src/collection/DefaultImage360Collection.ts [hunk 2]
```diff
@@ -44,6 +46,7 @@ export class DefaultImage360Collection implements Image360Collection {
   private readonly _icons: IconCollection;
   private _isCollectionVisible: boolean;
   private readonly _collectionId: string;
+  private readonly _styleAssignmentTracker: AnnotationStyleTracker = new AnnotationStyleTracker();
```
Comment (christjt) [inline viewer/packages/360-images/src/collection/DefaultImage360Collection.ts:49]: Any reason to not initialize this is the constructor?

File: viewer/packages/360-images/src/annotation-styling/AnnotationStyleTracker.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
+/*!
+ * Copyright 2023 Cognite AS
+ */
+
+import { AnnotationModel } from '@cognite/sdk/dist/src';
```
Comment (christjt) [inline viewer/packages/360-images/src/annotation-styling/AnnotationStyleTracker.ts:5]:
```suggestion
import { AnnotationModel } from '@cognite/sdk';
```

**Review by christjt — CHANGES_REQUESTED** [review]
Looks great, lets see if we can simplify the API a bit.

File: viewer/packages/360-images/src/annotation-styling/AnnotationStyleTracker.ts (hunk 1 above)
Comment (haakonflatval-cognite) [reply viewer/packages/360-images/src/annotation-styling/AnnotationStyleTracker.ts]: Outdated, removed `AnnotationStyleTracker`

File: viewer/packages/360-images/src/collection/DefaultImage360Collection.ts (hunk 2 above)
Comment (haakonflatval-cognite) [reply viewer/packages/360-images/src/collection/DefaultImage360Collection.ts]: Outdated, removed `AnnotationStyleTracker`

**Review by christjt — APPROVED** [review]
NIiiiiiiiice!

**Pull request merged**